mitmproxy/controller.py

```diff
@@ -239,52 +241,70 @@ class Reply(object):
     def __init__(self, obj):
         self.obj = obj
         self.q = queue.Queue()
-        # Has this message been acked?
-        self.acked = False
-        # Has the user taken responsibility for ack-ing?
-        self.taken = False
-        # Has a handler taken responsibility for ack-ing?
-        self.handled = False
+
+        self.state = "unhandled"  # "unhandled" -> "handled" -> "taken" -> "committed"
```

Comment:
Should we make a setter for `state` to verify the value? 
I don't like depending on some (possible random) string.